test/DynamoCoreTests/Migration/WorkspaceMigrationsTests.cs

```diff
@@ -15,21 +16,31 @@ class WorkspaceMigrationsTests : DynamoModelTestBase
         [Category("UnitTests")]
         public void TestWorkspaceMigrations()
         {
+            string consoleOutput;
+
             //Arrange
             string openPath = Path.Combine(TestDirectory, @"core\NodeStates.dyn");
             RunModel(openPath);
             string logFileText = string.Empty;
             XmlDocument xmlDoc = new XmlDocument();
 
-            //Act
-            //Create a instance of WorkspaceMigration and call the function that writes to the log
-            var workMigration = new WorkspaceMigrations(CurrentDynamoModel);
-            workMigration.Migrate_0_5_3_to_0_6_0(xmlDoc);
-            var currentLoggerPath = CurrentDynamoModel.Logger.LogPath;
+            using (StringWriter stringWriter = new StringWriter())
+            {
+                Console.SetOut(stringWriter);
+
+                //Act
+                //Create a instance of WorkspaceMigration and call the function that writes to the log
+                var workMigration = new WorkspaceMigrations(CurrentDynamoModel);
+                workMigration.Migrate_0_5_3_to_0_6_0(xmlDoc);
+
+                consoleOutput = stringWriter.ToString();
+
+                StreamWriter sw = new StreamWriter(Console.OpenStandardOutput());
+                Console.SetOut(sw);
+            }
 
-            //Assert
-            Assert.IsTrue(File.Exists(currentLoggerPath));//Check that the log was created successfully
-            Assert.IsTrue(CurrentDynamoModel.Logger.LogText.Contains("migration from 0.5.3.x to 0.6.1.x"));//Checks that the log text contains the migration info
+            //Checks that the console output text contains the migration info
+            Assert.IsTrue(consoleOutput.Contains("migration from 0.5.3.x to 0.6.1.x"));
```

Comment:
Maybe for this test we actually want to use the non-test version of the logger? I'm thinking that having a persistent log after a migration is done is desired and should be tested to exist.

Reply:
The thing is, the dynamo logger is created from the dynamo model when the dynamo model is initialized from the tests. It is a read only property and we create a instance of logger in non test mode here and add it to the model